src/primitives/compression.cpp

```diff
@@ -0,0 +1,283 @@
+#include <primitives/compression.h>
+
+CCompressedOutPoint::CCompressedOutPoint(const uint32_t block_height, const uint32_t block_index) : m_block_height(block_height), m_block_index(block_index), m_vout(0) {};
+
+CCompressedOutPoint::CCompressedOutPoint(const uint256& txid, const uint32_t vout) : m_block_height(0), m_block_index(0), m_txid(txid), m_vout(vout) {};
+
+std::string CCompressedOutPoint::ToString() const
+{
+    return strprintf("CCompressedOutPoint(block_height=%u, block_index=%u, txid=%s, vout=%u)",
+    m_block_height,
+    m_block_index,
+    HexStr(m_txid),
+    m_vout);
+}
+
+CCompressedTxIn::CCompressedTxIn(const CTxIn& txin, const CCompressedOutPoint& out, const CScript& script_pubkey) : m_prevout(out) {
+
+    m_compressedSignature = false;
+    m_hashType = 0;
+    m_isHashStandard = false;
+
+    if ((txin.scriptSig.size() || txin.scriptWitness.stack.size()) && (out.vout() != 0xFFFFFFFF))  {
+        CScript scriptSig;
+
+        if (script_pubkey.IsPayToTaproot() && txin.scriptWitness.stack.size() == 1) {
```

Comment:
In d86bd420e798449b2796ca49b2bd924ad83c658a:

I would prefer you switch on the `TxoutType` output of `Solver`, or even extract that logic out of `Solver` into its own utility function. (Fine to not do this, just call `Solver`, and leave a comment saying that you maybe should do the splitting.)

Reply:
Fixed, Thanks!